File: admin-app/CrepesCoffeeAdmin/Models/PedidoDetalle.cs
using System;
using Newtonsoft.Json;

namespace CrepesCoffeeAdmin.Models
{
    public class PedidoDetalle
    {
        [JsonProperty("id")]
        public int Id { get; set; }

        [JsonProperty("pedido_id")]
        public int PedidoId { get; set; }

        [JsonProperty("producto_id")]
        public int ProductoId { get; set; }

        [JsonProperty("producto")]
        public Producto? Producto { get; set; }

        [JsonProperty("cantidad")]
        public int Cantidad { get; set; }

        [JsonProperty("precio_unitario")]
        public decimal PrecioUnitario { get; set; }

        [JsonProperty("created_at")]
        public DateTime CreatedAt { get; set; }

        [JsonProperty("updated_at")]
        public DateTime UpdatedAt { get; set; }

        public decimal Subtotal => Cantidad * PrecioUnitario;
        public string SubtotalFormateado => $"${Subtotal:F2}";
    }
} 

File: backend/app/Models/PedidoDetalle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class PedidoDetalle extends Model
{
    use HasFactory;

    protected $fillable = [
        'pedido_id',
        'producto_id',
        'cantidad',
        'precio_unitario',
    ];

    protected $casts = [
        'precio_unitario' => 'decimal:2',
    ];

    /**
     * Obtener el pedido al que pertenece este detalle
     */
    public function pedido()
    {
        return $this->belongsTo(Pedido::class);
    }

    /**
     * Obtener el producto de este detalle
     */
    public function producto()
    {
        return $this->belongsTo(Producto::class);
    }
}
